modules/tensor_mechanics/src/materials/CreepStressUpdateBase.C

```diff
@@ -0,0 +1,66 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CreepStressUpdateBase.h"
+
+template <>
+InputParameters
+validParams<CreepStressUpdateBase>()
+{
+  InputParameters params = validParams<RadialReturnStressUpdate>();
+
+  params.addCoupledVar("temperature", 0.0, "Coupled temperature");
+  params.addDeprecatedParam<std::string>(
+      "creep_prepend",
+      "",
+      "String that is prepended to the creep_strain Material Property",
+      "This has been replaced by the 'base_name' parameter");
+  params.set<std::string>("effective_inelastic_strain_name") = "effective_creep_strain";
+
+  return params;
+}
+
+CreepStressUpdateBase::CreepStressUpdateBase(const InputParameters & parameters)
```

Comment:
Do you think it makes sense to name this `J2CreepStressUpdateBase` or something, just to make it clear that it is only applicable to J2 models?

Reply:
Isn't that already decided by RadialReturnStressUpdate?

Reply:
Sure. I'm just thinking that someone with a non-J2 model might pick this up and think they can derive their model from it. They ought to quickly see that they can't, but the name you have implies that it's more general than it is.

Reply:
Maybe `RadialReturnCreepStressUpdateBase`? Not a super pretty name, but it doesn't matter too much since it's not exposed in the input files.

Reply:
Yeah, I like `RadialReturnCreepStressUpdateBase` better, it makes is clear. A mouthful, but very clear.

Reply:
We had a very heated discussion about the `Update` in the name. The idea was to make it clear that these models couldn't compute the stress by themselves. 

What I'm still lobbying for is getting rid of `Compute` in front of all the models!

Reply:
Ah! I thought it was to separate it with solid mechanics.